caravel/views.py

```diff
@@ -1171,6 +1171,9 @@ def get_viz(
             args=None,
             datasource_type=None,
             datasource_id=None):
+
+        slice_id = slice_id or request.args.get('slice_id')
```

Comment:
we should keep `request` out of `models.py` and only in `views.py`, this logic belongs in methods that call this method.